### ISSUE TITLE
no error when class with `abstractmethod` doesn't extend `ABC`

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#12891]
```py
from abc import abstractmethod

class A:
    @abstractmethod
    def foo(self) -> int: ...

foo = A() # mypy error, but no runtime error
```
https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-context%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cno-implicit-optional%2Cno-implicit-reexport%2Clocal-partial-types%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=d0e7b9f621ee1cb69a25cae1549a6637

i think mypy should enforce extending `ABC`, as it throws a runtime error if an abstract class ever gets instanciated, which is slightly safer than relying entirely on mypy to prevent it

**Comment by AlexWaygood** [comment]
This is a feature, not a bug, and it's used extensively by typeshed. I wouldn't be surprised if it's used by other projects as well.

There's lots of classes in the standard library that have "implicitly abstract" methods. These are cases where the class doesn't have `ABCMeta` as the metaclass, and the method isn't decorated with `@abstractmethod`, but the runtime nonetheless expects you to override the method in a subclass, or things won't work as expected at runtime. Many of these classes were originally written before the introduction of the `abc` module, but some are newer -- there are a lot in `asyncio`. Having `ABCMeta` as the metaclass adds overhead to the runtime, so it's not always the best choice.

In typeshed, when we're writing stubs for these classes, we want to decorate these "implicitly abstract" methods with `@abstractmethod` in order to warn users that they need to override them in subclasses if things are going to work as expected. But we _don't_ want to lie and say that `ABCMeta` is the metaclass when it isn't. Doing that would cause mypy to raise spurious errors about metaclass conflict in cases of multiple inheritance, and have other unfortunate side effects as well. So it's very useful for us that mypy recognises methods decorated with `@abstractmethod` as abstract, even in situations where `ABCMeta` is not the metaclass.

TL;DR: feature not a bug.
